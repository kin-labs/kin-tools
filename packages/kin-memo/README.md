# @kin-tools/kin-memo

Library for creating and parsing KIN Memos.
