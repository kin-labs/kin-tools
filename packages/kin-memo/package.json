{
  "name": "@kin-tools/kin-memo",
  "version": "1.7.1",
  "license": "MIT"
}
