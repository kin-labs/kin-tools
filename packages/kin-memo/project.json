{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/kin-memo/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/kin-memo",
        "tsConfig": "packages/kin-memo/tsconfig.lib.json",
        "packageJson": "packages/kin-memo/package.json",
        "main": "packages/kin-memo/src/index.ts",
        "assets": ["packages/kin-memo/*.md"]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/kin-memo/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/packages/kin-memo"],
      "options": {
        "jestConfig": "packages/kin-memo/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
