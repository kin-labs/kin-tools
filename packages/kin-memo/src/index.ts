export * from './lib/create-kin-memo';
export * from './lib/kin-memo';
export * from './lib/kin-memo-constants';
export * from './lib/transaction-type';
