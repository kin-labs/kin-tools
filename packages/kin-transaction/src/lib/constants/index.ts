export const KIN_MINT_MAINNET = 'kinXdEcpDQeHPEuQnqmUgtYykqKGVFq6CeVX5iAHJq6';
export const MEMO_V1_TOKEN_ID = 'Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo';
