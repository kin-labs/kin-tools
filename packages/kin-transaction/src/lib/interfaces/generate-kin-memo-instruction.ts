import { TransactionType } from '@kin-tools/kin-memo';

export interface GenerateKinMemoInstruction {
  appIndex: number;
  type: TransactionType;
}
