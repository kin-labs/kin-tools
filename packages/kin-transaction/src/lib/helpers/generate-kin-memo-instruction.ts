import { createKinMemo } from '@kin-tools/kin-memo';
import { TransactionInstruction } from '@solana/web3.js';
import { GenerateKinMemoInstruction } from '../interfaces';
import { generateMemoInstruction } from './generate-memo-instruction';

/**
 * Method to format a correct Kin Memo Instruction based on appIndex and Type
 * @param {number} appIndex
 * @param {TransactionType} type
 * @returns {TransactionInstruction}
 */
export function generateKinMemoInstruction({
  appIndex,
  type,
}: GenerateKinMemoInstruction): TransactionInstruction {
  // Create correctly formatted memo string, including your App Index
  const memoContent = createKinMemo({ appIndex, type });
  // Create Memo Instruction for KRE Ingestion - Must be Memo Program v1, not v2
  return generateMemoInstruction({ memoContent });
}
