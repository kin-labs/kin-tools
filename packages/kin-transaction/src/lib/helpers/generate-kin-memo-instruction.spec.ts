import { TransactionType } from '@kin-tools/kin-memo'
import { MEMO_V1_TOKEN_ID } from '../constants'
import { generateKinMemoInstruction } from './generate-kin-memo-instruction'

describe('generateMemoInstruction', () => {
  it('should create a Memo Instruction', async () => {
    const memo = generateKinMemoInstruction({
      appIndex: 1,
      type: TransactionType.Spend,
    });

    expect(memo.programId.toBase58()).toEqual(MEMO_V1_TOKEN_ID);
    expect(memo).toMatchSnapshot();
  });
});
