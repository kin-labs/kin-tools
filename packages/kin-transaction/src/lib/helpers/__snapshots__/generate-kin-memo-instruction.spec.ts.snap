// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateMemoInstruction should create a Memo Instruction 1`] = `
TransactionInstruction {
  "data": Object {
    "data": Array [
      82,
      81,
      81,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      65,
      61,
    ],
    "type": "Buffer",
  },
  "keys": Array [],
  "programId": "Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo",
}
`;
